src/game/game.cpp

```diff
@@ -2517,12 +2525,16 @@ void Game::playerEquipItem(uint32_t playerId, uint16_t itemId, bool hasTier /* =
 	if (slotItem && slotItem->getID() == it.id && (!it.stackable || slotItem->getItemCount() == 100 || !equipItem)) {
 		internalMoveItem(slotItem->getParent(), player, CONST_SLOT_WHEREEVER, slotItem, slotItem->getItemCount(), nullptr);
 	} else if (equipItem) {
-		if (it.weaponType == WEAPON_AMMO) {
-			Item* quiver = player->getInventoryItem(CONST_SLOT_RIGHT);
-			if (quiver && quiver->isQuiver()) {
-				internalMoveItem(equipItem->getParent(), quiver->getContainer(), 0, equipItem, equipItem->getItemCount(), nullptr);
+		if (slotItem && slotItem->isQuiver()) {
+			if (it.weaponType == WEAPON_AMMO) {
+				internalMoveItem(equipItem->getParent(), slotItem->getContainer(), 0, equipItem, equipItem->getItemCount(), nullptr);
```

Comment:
Is it not good to validate if the internalMoveItem is returning RETURNVALUE_NOERROR?
```suggestion
				if (auto ret = internalMoveItem(equipItem->getParent(), slotItem->getContainer(), 0, equipItem, equipItem->getItemCount(), nullptr);
					ret != RETURNVALUE_NOERROR)
				{
					SPDLOG_ERROR("[{}] - Failed to move item to quiver");
					return;
				}
```